apps/web-shop-front/src/pages/CheckoutPage.tsx

```diff
@@ -0,0 +1,182 @@
+import { Button, Input, Loader, Image } from '@team21/ui-components'
+import useGetCartProducts from '@team21/web-shop-front/src/graphql/cart/useGetCartProducts';
+import { CartItem } from "@team21/types" 
+import { useUser } from '@team21/web-shop-front/src/state/state';
+import { useForm } from 'react-hook-form';
+import useCreateOrder from '@team21/web-shop-front/src/graphql/order/useCreateOrder'
+import useCreateOrderProducts from '@team21/web-shop-front/src/graphql/order/useCreateOrderProducts'
+import useRemoveItemFromCart from '@team21/web-shop-front/src/graphql/cart/useRemoveItemFromCart'
+
+const CheckoutPage: FC = () => {
+    const [{ user }] = useUser();
+    const { data, loading } = useGetCartProducts(user.cart_id);
+    const { register, handleSubmit, formState } = useForm();
+    const { errors } = formState;
+    const [createOrder] = useCreateOrder();
+    const [createOrderProducts] = useCreateOrderProducts();
+    const [removeItemFromCart] = useRemoveItemFromCart();
+    const [isLoader, setIsLoader] = useState(false);
+
+    if(!data && loading) return <Loader />
+    if(isLoader) return <Loader />
+
+    const cartItems = data?.cart_item
+
+    let orderTotal = 0;
+    cartItems?.map((item: CartItem) => {
+        orderTotal += item.product.price * item.quantity
+    })
+
+    async function continueToPayment (data: { first_name: string; last_name: string; country: string; city: string; address: string; zip_code: string; phone: string; }){
+        setIsLoader(true);
+        const response = await createOrder({ variables: {
+            first_name: data.first_name,
+            last_name: data.last_name,
+            country: data.country,
+            city: data.city,
+            address: data.address,
+            zip_code: data.zip_code,
+            phone: data.phone,
+            user_id: user.id,
+            total_price: orderTotal
+        } })
+        const orderId = response.data.insert_orders_one.id
+        if(orderId && cartItems){
+            for(const item of cartItems){
+                await createOrderProducts({ variables: {
+                    order_id: orderId,
+                    price: item.product.price,
+                    product_id: item.product.id,
+                    quantity: item.quantity
+                }})
+                await removeItemFromCart({ variables: {
+                    id: item.id
+                }})
+            }
+        }
+        setIsLoader(false);
+    }
+
+    return <div className="w-10/12 p-24 mx-auto">
+            <h1 className="text-h1 mb-32"> Checkout </h1>
+            <div className="flex justify-between">
+                <div className="flex flex-col w-2/5 px-40">
+                    <h2 className="text-h3 font-medium text-left mb-24">Contact information</h2>
+                    <form onSubmit={handleSubmit(continueToPayment)}>
+                        <Input 
+                            id="first_name" 
+                            type="text" 
```

Comment:
you can maybe extract this form to new component or something like that, so we don't have too long files.. but we can do it later